src/ray/gcs/gcs_server/gcs_task_manager.cc

```diff
@@ -423,6 +423,28 @@ void GcsTaskManager::HandleGetTaskEvents(rpc::GetTaskEventsRequest request,
       return false;
     }
 
+    if (filters.has_state()) {
+      const google::protobuf::EnumDescriptor *task_status_descriptor =
+          ray::rpc::TaskStatus_descriptor();
+
+      // Figure out the latest state of a task.
+      ray::rpc::TaskStatus state = ray::rpc::TaskStatus::NIL;
+      if (task_event.has_state_updates()) {
```

Comment:
since we are also doing something similar at the client when aggregating later, i wonder if this would go out of sync with that routine. 

I wonder if it makes sense to add a `current_state` to the task event when returned (even when there's no state filter), so we don't need to do that at the client. 

This probably could go in another PR.

Reply:
Yea, we could do that. But based on the current logic, it shouldn't go out of sync since we just iterate through TaskStatus and find the last state that has timestamp. 

Having a `current_state` or define the common function in cython is a good follow-up to do.